deepreg/model/loss/image.py

```diff
@@ -286,14 +210,12 @@ def call(self, y_true: tf.Tensor, y_pred: tf.Tensor) -> tf.Tensor:
             y_pred = tf.expand_dims(y_pred, axis=4)
         assert len(y_true.shape) == len(y_pred.shape) == 5
 
-        filters, kernel_vol = self.kernel_fn(
+        filters = self.kernel_fn(
             kernel_size=self.kernel_size,
-            input_channel=y_true.shape[4],
         )
+        kernel_vol = tf.reduce_sum(filters) ** 3
```

Comment:
r u sure here the filters have all values = 1, maybe https://www.tensorflow.org/api_docs/python/tf/ones_like would be helpful?

Reply:
@YipengHu @kate-sann5100 cc

Reply:
This is the deduction process of variance, in case anyone is interested:
![image](https://user-images.githubusercontent.com/44606435/104332426-110ad300-54e8-11eb-8fde-859f7a6f33f1.png)

Reply:
looks good to me (need to define you \bar{b})

Reply:
I believe

- the calculation of `kernel_vol` for triangular and gaussian kernels is wrong in both the main branch and the current merge request. 
- `kernel_vol` should equal to the sum of all elements in the 3d kernel (matrix multiplication results of all separable kernels).
- @mathpluscode 's implementation should be correct.

Reply:
@acasamitjana can you have a look at this with @kate-sann5100 and @mathpluscode ? if confirmed, consider to change this ticket to a bug-fixing one.